kumo/climate.py

```diff
@@ -95,18 +160,43 @@ def temperature_unit(self):
     @property
     def current_humidity(self):
         """Return the current humidity, if known."""
-        humidity = self._pykumo.get_current_humidity()
-        return humidity
+        return self._current_humidity
+
+    def _update_current_humidity(self):
+        """Refresh cached current humidity."""
+        self._current_humidity = self._pykumo.get_current_humidity()
 
     @property
     def hvac_mode(self):
-        """Return current hvac operation mode."""
+        """Return current hvac operation state."""
+        return self._hvac_mode
+
+    def _update_hvac_mode(self):
+        """Refresh cached hvac mode."""
         mode = self._pykumo.get_mode()
         try:
             result = KUMO_STATE_TO_HA[mode]
         except KeyError:
-            result = HVAC_MODE_OFF
-        return result
+            result = None
+        self._hvac_mode = result
+
+    @property
+    def hvac_action(self):
+        """Return current hvac operation in action."""
+        return self._hvac_action
+
+    def _update_hvac_action(self):
+        """Refresh cached hvac action"""
+        standby = self._pykumo.get_standby()
```

Comment:
I'm going to need your mods to pykumo for this change

Reply:
Just sent a pull request with that method.